odxtools/diaglayercontainer.py

```diff
@@ -71,26 +56,26 @@ def from_et(et_element: ElementTree.Element) -> "DiagLayerContainer":
         description = create_description_from_et(et_element.find("DESC"))
         admin_data = AdminData.from_et(et_element.find("ADMIN-DATA"), doc_frags)
         company_datas = create_company_datas_from_et(et_element.find("COMPANY-DATAS"), doc_frags)
-        ecu_shared_datas = [
+        ecu_shared_datas = NamedItemList(short_name_as_id, [
             DiagLayer.from_et(dl_element, doc_frags)
             for dl_element in et_element.iterfind("ECU-SHARED-DATAS/ECU-SHARED-DATA")
-        ]
-        protocols = [
+        ])
+        protocols = NamedItemList(short_name_as_id, [
             DiagLayer.from_et(dl_element, doc_frags)
             for dl_element in et_element.iterfind("PROTOCOLS/PROTOCOL")
-        ]
-        functional_groups = [
+        ])
+        functional_groups = NamedItemList(short_name_as_id, [
             DiagLayer.from_et(dl_element, doc_frags)
             for dl_element in et_element.iterfind("FUNCTIONAL-GROUPS/FUNCTIONAL-GROUP")
-        ]
-        base_variants = [
+        ])
+        base_variants = NamedItemList(short_name_as_id, [
             DiagLayer.from_et(dl_element, doc_frags)
             for dl_element in et_element.iterfind("BASE-VARIANTS/BASE-VARIANT")
-        ]
-        ecu_variants = [
+        ])
+        ecu_variants = NamedItemList(short_name_as_id, [
```

Comment:
is short_name_as_id actually needed anymore?

Reply:
yes: without it, `NamedItemList` does not know how to map an item to a name. also, ODX short names can start with a number, so we cannot simply pass through the short name a la `lambda x: x.short_name`...

Reply:
do you have actually a case where NamedItemList have another id mapper other than short_name_as_id?

Reply:
not within odxtools. (I think that this class ought to be implemented generically, IMO). To make its usage a bit more convenient, we could introduce a `ShortNameItemList` class for this purpose, though...